src/components/Products/ProductDetails/ProductDetails.jsx

```diff
@@ -0,0 +1,114 @@
+import { Icon } from "@iconify/react";
+import { useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+import { useData } from "../../../context";
+import { AddToCart } from "../../Cart/Add-To-Cart";
+import Footer from "../../Layout/Footer";
+import { getProductBasedOnId } from "./../../../services/index"
+
+const ProductDetails = () => {
+    const { id } = useParams();
+    const { state: { oneProduct }, dispatch } = useData();
+
+    useEffect(() => {
+        (async () => {
+            await getProductBasedOnId(dispatch, id);
+        })();
+    }, [dispatch])
+    useEffect(() => {
+        document.title = "APNA SHOP | Product Details";
```

Comment:
this is an anti-pattern right You can use `<Route title="APNA SHOP | Product Details" .../>` or **Helmet** package

Reply:
Goog to Know Amin I'll go with that